packages/react-next/src/components/Coachmark/Coachmark.base.tsx

```diff
@@ -116,36 +118,15 @@ function useBeakPosition(
   targetAlignment: RectangleEdge | undefined,
   targetPosition: RectangleEdge | undefined,
 ) {
-  const beakDirection = targetPosition === undefined ? RectangleEdge.bottom : getOppositeEdge(targetPosition);
+  const { theme } = props;
 
-  /**
-   * The left position of the beak
-   */
-  const [beakLeft, setBeakLeft] = React.useState<string>();
+  return React.useMemo(() => {
```

Comment:
The parts of the beak position will never change independently, so this is a much more efficient way to calculate and update them.